src/javascript/ContentEditor.context.js

```diff
@@ -53,6 +63,7 @@ export const withContentEditorDataContextProvider = (formQuery, formDataAdapter)
             return <ProgressOverlay/>;
         }
 
+        sections = sections.length === 0 ? formSections : sections;
```

Comment:
The setSection has been added because the context was refreshed but should not. 
https://jira.jahia.org/browse/BACKLOG-13837 should work on that